llvm/lib/Transforms/Scalar/LoopFlatten.cpp

```diff
@@ -343,7 +343,7 @@ static bool verifyTripCount(Value *RHS, Loop *L,
     // If the RHS of the compare is equal to the backedge taken count we need
     // to add one to get the trip count.
     if (SCEVRHS == BackedgeTCExt || SCEVRHS == BackedgeTakenCount) {
-      ConstantInt *One = ConstantInt::get(ConstantRHS->getType(), 1);
+      ConstantInt *One = ConstantInt::get(ConstantRHS->getIntegerType(), 1);
```

Comment:
Why are you changing usages like these? This code should work fine with `getType()`.

Reply:
Much the same reason as with `getBitWidth()` but the need for immediate change is greater so I renamed this method so changes to existing code paths are minimal whilst still providing a route to trigger asserts once testing is expanded.

I did consider just removing the method and adding the necessary casts but figured somebody went to the trouble of adding the override in the first place so I maintained this but under a modified name. Do you think the shorthand is not worth it?

Reply:
I don't really follow here. Even if you rename the overload on ConstantInt, there will still be the method inherited from `Value::getType()`, and using that method here should work (and be forward-compatible with vector ConstantInt), because `ConstantInt::get` doesn't actually require that the argument is an IntegerType.

Reply:
Could it be `static ConstantInt *get(IntegerType *Ty, uint64_t V, bool IsSigned = false);`?

I don't think I made this change up.  I wanted a mechanical change so just removed the overload and the compiler told me all the places that relied on it.

Reply:
Ooooh, I think the issue is that it's stored in a `ConstantInt *` variable. Changing it to `Constant *` should make it pick the other overload, I think.